src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx

```diff
@@ -0,0 +1,204 @@
+    ) {
+        const { dataSource } = options;
+        if (dataSource) {
+            const {
+                dimensions: { height },
+                custom: { drillableItems },
+                locale,
+                config,
+            } = options;
+            const {
+                afterRender,
+                onDrill,
+                onFiredDrillEvent,
+                onError,
+                onExportReady,
+                onLoadingChanged,
+            } = this.callbacks;
+
+            // keep height undef for AD; causes indigo-visualizations to pick default 100%
+            const resultingHeight = this.environment === DASHBOARDS_ENVIRONMENT ? height : undefined;
+            const allProperties: IVisualizationProperties = get(
+                visualizationProperties,
+                "properties",
+                {},
+            ) as IVisualizationProperties;
+            const resultSpecWithDimensions: AFM.IResultSpec = {
+                ...options.resultSpec,
+                dimensions: this.getDimensions(mdObject),
+            };
+            const enableSortingByTotalGroup = this.featureFlags.enableSortingByTotalGroup as boolean;
+            const sorts: AFM.SortItem[] = createSorts(
+                this.type,
+                dataSource.getAfm(),
+                resultSpecWithDimensions,
+                allProperties,
+                canSortStackTotalValue(mdObject, null, enableSortingByTotalGroup),
+                enableSortingByTotalGroup,
+            );
+            const resultSpecWithSorts = {
+                ...resultSpecWithDimensions,
+                sorts,
+            };
+            const fullConfig = this.buildVisualizationConfig(mdObject, config, null);
+            const geoPushpinProps = {
+                projectId: this.projectId,
+                drillableItems,
+                onDrill,
+                onFiredDrillEvent,
+                config: fullConfig,
+                height: resultingHeight,
+                locale,
+                dataSource,
+                resultSpec: resultSpecWithSorts,
+                afterRender,
+                onLoadingChanged,
+                pushData: this.handlePushData,
+                onError,
+                onExportReady,
+                LoadingComponent: null as any,
+                ErrorComponent: null as any,
+                intl: this.intl,
+            };
+
+            render(<GeoChart {...geoPushpinProps} />, document.querySelector(this.geoPushpinElement));
+        }
+    }
+
+    private getMeasuresSizeBucketItem(buckets: IBucket[]): IBucketItem[] {
+        if (hasBucket(buckets, BucketNames.MEASURES)) {
+            return getBucketItemsByType(buckets, BucketNames.MEASURES, [METRIC]).slice(0, 1);
+        }
+
+        return getBucketItemsByType(buckets, BucketNames.SIZE, [METRIC]);
+    }
+
+    private getMeasuresColorBucketItem(buckets: IBucket[]): IBucketItem[] {
```

Comment:
We only have one measure in bucket so I think `getMeasureColorBucketItem ` is fine.